core/src/View/Kokkos_ViewAlloc.hpp

```diff
@@ -188,6 +188,54 @@ struct ViewValueFunctor {
   }
 };
 
+template <class DeviceType, class ValueType>
+struct ViewValueFunctorHostSerialInit {
+  using ExecSpace = typename DeviceType::execution_space;
```

Comment:
I didn't find a check that the execution space can access host space anywhere else.
```suggestion
  using ExecSpace = typename DeviceType::execution_space;
  static_assert(Kokkos::SpaceAccessibility<ExecSpace, Kokkos::HostSpace>::accessible);
```

Reply:
Yeah looks like we just pass the View device type in. But this initialization scheme should be valid for any View whose memory space is host accessible.

Reply:
> this initialization scheme should be valid for any View whose memory space is host accessible

See L195